standards/nestjs_standards.md

```diff
@@ -0,0 +1,372 @@
+# NestJS Standards Guide
+
+This comprehensive guide outlines the recommended standards and best practices for developing and working with NestJS applications. Following these guidelines will help you build maintainable, scalable, and efficient server-side applications.
+
+## Project Structure
+
+### Recommended Directory Layout
+
+```text
+src/
+├── modules/              # Feature modules
+│   ├── users/            # Example feature module
+│   │   ├── dto/          # Data Transfer Objects
+│   │   ├── entities/     # Database entities
+│   │   ├── interfaces/   # Type definitions
+│   │   ├── users.controller.ts
+│   │   ├── users.module.ts
+│   │   ├── users.service.ts
+│   │   └── users.spec.ts
+├── common/              # Shared utilities
+│   ├── filters/         # Exception filters
+│   ├── interceptors/    # Request/response interceptors
+│   ├── guards/          # Auth guards
+│   ├── pipes/           # Validation pipes
+│   └── decorators/      # Custom decorators
+├── config/              # Configuration files
+├── main.ts              # Application entry point
+└── app.module.ts        # Root module
+```
+
+### Module Organization Principles
+
+- Each feature in its own module; cross-cutting concerns (auth, email, metrics) go in common or shared modules.
+- Small apps: flat structure; larger apps: domain or hexagonal architecture per module.
+- Use barrel files for clean imports—avoid hidden dependencies.
+- Use `forwardRef()` judiciously to resolve circular references.
+
+## Coding Standards
+
+### General Guidelines
+
+```ts
+// Example: Async/Await
+async function fetchData() {
+  const result = await this.httpService
+    .get("https://api.example.com")
+    .toPromise();
+  return result.data;
+}
+```
+
+- Follow **TypeScript** conventions and leverage strong typing
+- Prefer `async/await` over Promises directly
+- Use **ESLint** and **Prettier** for consistent style enforcement
+- Avoid magic strings/numbers — use enums or constants
+
+```ts
+// Example: Constants
+export const DEFAULT_ROLE = "user";
+```
+
+- Name files consistently (`.controller.ts`, `.service.ts`, etc.)
+
+### Naming Conventions
+
+- `camelCase` for variables and functions
+- `PascalCase` for classes and interfaces
+- `snake_case` for database columns (if applicable)
+- `UPPER_CASE` for environment variables
+
+### Service & Controller Patterns
+
+```ts
+// users.controller.ts
+@Controller("users")
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Get()
+  getAllUsers() {
```

Comment:
We should always have Authorization in place per endpoint. I think we're using CASL for permission-based